### ISSUE TITLE
How to modify number of trainning epochs?

### ISSUE THREAD

**Issue by Mr-Doyle** [MIC-DKFZ/nnUNet#1389]
Hello! I had a problem when using nnUNetV2: when trainning models on my own datasets, model performance (loss and Dice)  remained stable when epoch was about 150, indicating the model converged already. I want to finish the training process then, but the default num_epoch is 1000, so it won't stop. I wonder how I can modify num_epoch to the certain value I want?

**Comment by amoreyo** [comment]
I have the same issue while I training my own dataset. I have tried to change the parameters in “ nnUNet/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py” 
‘’’
        self.initial_lr = 1e-2
        self.weight_decay = 3e-5
        self.oversample_foreground_percent = 0.33
        self.num_iterations_per_epoch = 25
        self.num_val_iterations_per_epoch = 5
        self.num_epochs = 100 // default 1000
        self.current_epoch = 0
‘’’
But it didn’t work.  I have no idea about this issue.  If you solve the problem plz let me know.

**Comment by HungTeen** [comment]
you can change `self.max_num_epochs = 1000 ` in the __init__ function of nnUNetTrainerV2, it works for me.

**Comment by FabianIsensee** [comment]
https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/variants/training_length/nnUNetTrainer_Xepochs.py

I do not recommend to mess with the nnunet defaults. Let it do its thing. And if you think you can do better, make changes AFTER that so that you can be sure not to degrade performance ;-)

**Comment by Mr-Doyle** [comment]
> you can change `self.max_num_epochs = 1000 ` in the **init** function of nnUNetTrainerV2, it works for me.

I also modified my code this way and it works too. Thanks~

**Comment by Mr-Doyle** [comment]
> https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/variants/training_length/nnUNetTrainer_Xepochs.py
> 
> I do not recommend to mess with the nnunet defaults. Let it do its thing. And if you think you can do better, make changes AFTER that so that you can be sure not to degrade performance ;-)

Wow, thank you so much!  I will look into this and try finding better configurations for my work. Thanks again! ^v^
